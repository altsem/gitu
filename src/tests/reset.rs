use super::*;

fn setup() -> TestContext {
    let ctx = TestContext::setup_clone();
    commit(ctx.dir.path(), "unwanted-file", "");
    ctx
}

#[test]
pub(crate) fn reset_menu() {
    snapshot!(setup(), "lljX");
}

#[test]
pub(crate) fn reset_soft_prompt() {
    snapshot!(setup(), "lljXsq");
}

#[test]
pub(crate) fn reset_soft() {
    snapshot!(setup(), "lljXs<enter>q");
}

#[test]
pub(crate) fn reset_mixed() {
    snapshot!(setup(), "lljXm<enter>q");
}

#[test]
fn reset_hard() {
    snapshot!(setup(), "lljXh<enter>q");
}
