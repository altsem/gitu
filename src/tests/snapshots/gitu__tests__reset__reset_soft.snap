---
source: src/tests/reset.rs
expression: ctx.redact_buffer()
---
▌On branch main                                                                 |
▌Your branch is up to date with 'origin/main'.                                  |
                                                                                |
 Staged changes (1)                                                             |
 added   unwanted-file                                                          |
                                                                                |
 Recent commits                                                                 |
 b66a0bf main origin/main add initial-file                                      |
                                                                                |
                                                                                |
                                                                                |
                                                                                |
                                                                                |
                                                                                |
                                                                                |
                                                                                |
                                                                                |
                                                                                |
                                                                                |
                                                                                |
styles_hash: 30afab145e7cf548
